main_server.py

```
@@ -402,6 +386,66 @@ async def get_response(self, path, scope):
+        initialize_character_data=initialize_character_data,
+    )
+
+@app.post('/api/beacon/shutdown')
+async def beacon_shutdown():
+    """Beacon API for graceful server shutdown"""
+    try:
+        # 从 app.state 获取配置
+        current_config = get_start_config()
+        # Only respond to beacon if server was started with --open-browser
+        if current_config['browser_mode_enabled']:
+            logger.info("收到beacon信号，准备关闭服务器...")
```

Ruff (0.14.8)

String contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF001)